Add test plans for lecturer interface (#114)
Gamify-IT/docs: dev-manuals/test-plans/lecturer-interface-frontend/plans/u-lecturer-interface-13-edit-minigame-configuration-chickenshock.md

# Lecturer-Interface: Edit minigame configuration of chickenshock (`u.lecturer-interface-13`)

Version: V1.0, 09.08.2022 \
Author: Max Kästner

## Description

> The chickenshock configuration of a minigame can be edited in a modal. For minigames of a world the path is `/courses/<course-id>/worlds/<world-index>/minigames` and for a dungeon `/courses/<course-id>/worlds/<world-index>/dungeons/<dungeon-index>/minigames`.

Reviewer: ```suggestion
A `chickenshock` configuration can be customized to display custom questions and answers.
```

## Precondition

> At least one course is created. The tester is on the view of an area minigames list as described in the description above.

Reviewer: ```suggestion
At least one course with a `CHICKENSHOCK` minigame is created. The tester is located where the `chickenshock` instance can be configured.
```

Author: Why should this be as precondition that the minigame is already a chickenshock game?

Reviewer: See one below, because you then delete the first four steps in the message below, and because this setup has nothing to do with the test itself.

## Postcondition

> A configuration of the a chickenshock minigame was changed.

Reviewer: ```suggestion
The configuration of the `chickenshock` instance was changed.
```

## Typical procedure

1. Click on a dropwdown of a minigame to change the game
2. Select `CHICKENSHOCK`
3. A feedback toast message appears which contains that game was updated

> 4. The game is now displayed in the dropdown

Reviewer: Can now be deleted.

5. Click the edit button where the game was updated
6. A modal opens
7. Press add question button
8. The modal changes

> 9. Fill out the form:
>     - Enter a question
>     - Enter a correct answer
>     - Enter a wrong answer
>         - click add
>     - Enter another answer
>         - click add

Reviewer: ```suggestion
9. Fill out the form, including the question, the correct answer, and two wrong answers
```

10. Press the Ok button of the modal

> 11. The modal changes back

Reviewer: ```suggestion
11. The modal gets closed
```

Author: Technically it gets closed and another opens, but the user gets just a content change displayed, so he does not really know this

Reviewer: Yes. And I am writing this from the perspective of a user trying to test it.

12. The entered question with correct answer and wrong answers is displayed in the modal
13. Press the Ok button of the modal
14. The modal closes

> 15. A feedback toast message appears which contains that configuraiton was saved

Reviewer: ```suggestion
15. A success message appears
```

## Alternative procedures

> 

Reviewer: Can't you cancel editing the configuration?

Author: Yes
